test/plugin_functional/plugins/data_search/server/plugin.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { CoreSetup, Plugin } from 'kibana/server';
+import { schema } from '@kbn/config-schema';
+import { PluginStart as DataPluginStart } from '../../../../../src/plugins/data/server';
+
+export interface DataSearchTestStartDeps {
+  data: DataPluginStart;
+}
+
+export class DataSearchTestPlugin
+  implements Plugin<TestPluginSetup, TestPluginStart, {}, DataSearchTestStartDeps> {
+  public setup(core: CoreSetup<DataSearchTestStartDeps>) {
+    const router = core.http.createRouter();
+
+    router.get(
+      { path: '/api/data_search_plugin/search_source/as_scoped', validate: false },
```

Comment:
These tests create a few dummy routes that can be called from the client to execute various pieces of the searchSource service on the server.